cmd/api/main.go

```diff
@@ -0,0 +1,19 @@
+package main
+
+import (
+	"ecommerce_order/internal/adapters/http"
+	"log"
+
+	"github.com/gin-gonic/gin"
+	"github.com/joho/godotenv"
+)
+
+func main() {
```

Comment:
```suggestion
type App struct {
    Router   *gin.Engine
    handlers Handlers
}

func main() {
    app := &App{
        Router: gin.Default(),
        handlers: Handlers{
            CreateOrderHandler: NewCreateOrderHandler(placeOrderUseCase),
        },
    }
    app.MapWebRoutes()

    app.Router.Run()
}

func (a *App) MapWebRoutes() {
    a.Router.POST("/orders", a.handlers.CreateOrderHandler)
}
```

Reply:
E cada handler já está com o Caso de Uso injetado:


```suggestion
func NewCreateOrderHandler(placeOrderUseCase PlaceOrderUseCase) gin.HandlerFunc {
    return func(ctx *gin.Context) {
        // Lógica do handler usando o caso de uso já recebido (e.g., placeOrderUseCase.Create(...))
    }
}
```